### ISSUE TITLE
How to apply on yolov3 any instructions?

### ISSUE THREAD

**Issue by lucasjinreal** [kemaloksuz/RankSortLoss#3]
How to apply on yolov3 any instructions?

**Comment by kemaloksuz** [comment]
Hi,

YOLO-based architectures are not currently the among the methods we trained with RS Loss, thus I cannot provide "instructions", but I can provide some opinions. If I were to try, then I would follow the following roadmap:

- Instead of predicting 4+1+C dimensional output for each anchor (4: for box regression, 1: for objectness, C: for class probabilities), I would modify the design such that the network predicts 4+C dim output because I expect that the objectness of YOLO could be removed since it is also prioritizes detections. I expect that our sorting loss handles objectness similar to centerness/IoU branches. However, difficult to be sure without experimental proof. 
- We always used binary Sigmoid classifiers for each class, instead of softmax over all classes. I think YOLO also prefers sigmoid but I recommend checking it.
- After removal of the objectness scores and the decision to use sigmoid classifiers, inference time implementation should be reviewed.
- For training, given 4+C dimensional output, RS Loss can be easily applied as we did for several architectures. You can check out Fig. 3 in our paper for details or some of our implementation in this repo. Basically, you need to incorporate RS Loss and GIoU Loss together with score-based instance weighting and value-based task weighting.
- Finally, learning rate needs to be tuned, and I think this is important since for all the networks we trained, the learning rate of RS Loss is different from the default learning rate.

I think it is worth to try, and please let me know if you do so. Thanks for your interest in our work. 

Feel free to get in touch if you have more questions.

**Comment by lucasjinreal** [comment]
@kemaloksuz Gracis amigo. I'll do it

**Comment by CangHaiQingYue** [comment]
> @kemaloksuz Gracis amigo. I'll do it

大佬。有没有成功呀~
Mogul. Did you succeed~
